Add instructions how to debug NativeAOT exceptions on the ARM64 (#44631)
dotnet/docs: docs/core/deploying/native-aot/diagnostics.md

The Native AOT compiler generates information about line numbers, types, locals, and parameters. The native debugger lets you inspect stack trace and variables, step into or over source lines, or set line breakpoints.

> To debug managed exceptions, set a breakpoint on the `RhThrowEx` method, which is called whenever a managed exception is thrown. The exception is stored in the first argument register that is `rcx` on x64 and `x0` on Arm64. If your debugger supports viewing C++ objects, you can cast

Reviewer: ```suggestion
To debug managed exceptions, set a breakpoint on the `RhThrowEx` method, which is called whenever a managed exception is thrown. The exception is stored in the first argument register, which is `rcx` on x64 and `x0` on Arm64. If your debugger supports viewing C++ objects, you can cast
```

To set a breakpoint that breaks whenever an exception is thrown, choose the **Breakpoints** option from the **Debug > Windows** menu. In the new window, select **New > Function** breakpoint. Specify `RhThrowEx` as the Function Name and leave the Language option at **All Languages** (don't select C#).

To see what exception was thrown, start debugging (**Debug > Start Debugging** or <kbd>F5</kbd>) and when the `RhThrowEx` breakpoint hits, open the Watch window (**Debug > Windows > Watch**), and add following expression as one of the watches:

> |                    | Expression                            |

Reviewer: ```suggestion
| Architecture       | Expression                            |
```

> To set a breakpoint that breaks whenever an exception is thrown, choose the **Breakpoints** option from the **Debug > Windows** menu. In the new window, select **New > Function** breakpoint. Specify `RhThrowEx` as the Function Name and leave the Language option at **All Languages** (don't select C#).

Reviewer 2: There is one more mention of register names above. Could you please clarify it as well?

The exception is stored in the `rcx` or `x0` register. -> The exception is stored in the first argument register that is `rcx` on x64 and `x0` on Arm64.

Author: Done

> To see what exception was thrown, start debugging (`Debug` -> `Start Debugging` or `F5`), open the Watches window (`Debug` -> `Windows` -> `Watch`) and add following expression as one of the watches: `(S_P_CoreLib_System_Exception*)@rcx` on x64, or `(S_P_CoreLib_System_Exception*)x0` on ARM64. This leverages the fact that at the time `RhThrowEx` is called, the registers RCX and X0 contain the thrown exception, on x64 and ARM64, respectively. You can also paste the expression into the `Immediate Window`; the syntax is the same as for watches.

Reviewer 3: What's the motivation behind the formatting changes?

Is the `(S_P_CoreLib_System_Exception*)x0` part missing a `@`?

Author: Honestly that was copypaste, because it looks similar. Let me check again.

Reviewer: Yes, the VS menu items should not be code fenced. They should be bold.

Author: I fix an issues with formatting.

> To see what exception was thrown, start debugging (**Debug > Start Debugging** or <kbd>F5</kbd>), open the Watch window (**Debug > Windows > Watch**), and add following expression as one of the watches: `(S_P_CoreLib_System_Exception*)@rcx` on x64, or `(S_P_CoreLib_System_Exception*)x0` on ARM64. This mechanism leverages the fact that at the time `RhThrowEx` is called, the x64 CPU register RCX or ARM64 register X0 contains the thrown exception, on x64 and ARM64, respectively. You can also paste the expression into the **Immediate Window**; the syntax is the same as for watches.

Reviewer 3: ```suggestion
To see what exception was thrown, start debugging (**Debug > Start Debugging** or <kbd>F5</kbd>) and when the `RhThrowEx` breakpoint hits, open the Watch window (**Debug > Windows > Watch**), and add following expression as one of the watches:

|                    | Expression                            |
|--------------------|---------------------------------------|
| x64 architecture   | `(S_P_CoreLib_System_Exception*)@rcx` |
| Arm64 architecture | `(S_P_CoreLib_System_Exception*)@x0`  |

This mechanism leverages the fact that at the time `RhThrowEx` is called, the CPU register mentioned in the table contains the thrown exception. You can also paste the expression into the Visual Studio **Immediate Window**; the syntax is the same as for watches.
```
